scripts/msmtpq/msmtpq

```diff
@@ -69,22 +69,28 @@ MSMTP=${MSMTP:-msmtp}
 ##            ( chmod 0700 msmtp.queue )
 ##
 ## the queue dir - export this variable to reflect where you'd like it to be  (no quotes !!)
-Q=${Q:-~/.msmtp.queue}
-[ -d "$Q" ] || mkdir -m 0700 -p "$Q" || \
-  err '' "msmtpq : can't find or create msmtp queue directory [ $Q ]" ''     # if not present - complain ; quit
+MSMTPQ_Q=${MSMTPQ_Q:-${Q:-~/.msmtp.queue}}
+[ -d "$MSMTPQ_Q" ] || mkdir -m 0700 -p "$MSMTPQ_Q" || \
+  err '' "msmtpq : can't find or create msmtp queue directory [ $MSMTPQ_Q ]" ''     # if not present - complain ; quit
 ##
 ## set the queue log file var to the location of the msmtp queue log file
 ##   where it is or where you'd like it to be
-##     ( note that the LOG setting could be the same as the )
+##     ( note that the MSMTPQ_LOG setting could be the same as the )
 ##     ( 'logfile' setting in .msmtprc - but there may be   )
 ##     ( some advantage in keeping the two logs separate    )
 ##   if you don't want the log at all set the var to an empty string
 ##     (doing so would be inadvisable under most conditions, however)
 ##
 ## the queue log file - export this variable to change where logs are stored  (but no quotes !!)
 ##                      Set it to "" (empty string) to disable logging.
-[ -v LOG ] || LOG=~/log/msmtp.queue.log
-[ -d "$(dirname "$LOG")" ] || mkdir -p "$(dirname "$LOG")"
+if [ ! -v MSMTPQ_LOG ] ; then
+  if [ -v LOG ] ; then
+    MSMTPQ_LOG="$LOG"
+  else
+    MSMTPQ_LOG=~/log/msmtp.queue.log
+  fi
+fi
```

Comment:
This could be a one-liner:
```sh
MSMTP_LOG=${MSMTP_LOG:-${LOG:-"~/log/msmtp.queue.log"}}
```

Reply:
Actually, you would have to remove the quotes. The quotes prevent `~` from being expanded to the user’s home directory:

```bash
$ unset MSMTPQ_LOG LOG
$ echo "${MSMTPQ_LOG:-${LOG:-"~/log/msmtp.queue.log"}}"
~/log/msmtp.queue.log
$ echo ${MSMTPQ_LOG:-${LOG:-~/log/msmtp.queue.log}}
/home/jayman/log/msmtp.queue.log
$
```

Additionally, using `${MSMTPQ_LOG:-default}` would change the behavior of `msmtpq`. `MSMTPQ_LOG=${MSMTPQ_LOG:-default}` would set `MSMTPQ_LOG` to its default value if `MSMTPQ_LOG` is unset:

```bash
$ unset MSMTPQ_LOG
$ echo "${MSMTPQ_LOG:-default}"
default
```

or if `MSMTPQ_LOG` is set to an empty string:

```bash
$ MSMTPQ_LOG=""
$ echo "${MSMTPQ_LOG:-default}"
default
```

The default value should only be used if `MSMTPQ_LOG` and `LOG` are both unset. Before the commit that renamed `LOG`, [`README.msmtpq` said](https://github.com/marlam/msmtp-mirror/blob/98ad6b90d0920867a4ffbc6cc11481dc4570c601/scripts/msmtpq/README.msmtpq#L76):

<blockquote><pre>the LOG variable should have the desired name & location of the queue
log ; set it to an empty string (export LOG="") to disable logging</pre></blockquote>

It might make sense to change that behavior (may be there could be a separate variable, `MSMTPQ_ENABLING_LOGGING`, that controls whether or not logging happens), but that should be the topic of a separate PR.